pkg/util/logger_windows.go

```diff
@@ -0,0 +1,134 @@
+	}
+	return &l
+}
+
+func makeWindowsEventLogger(cfg *server.Config) (log.Logger, error) {
+	// error is necessary for the Log function to return an error
+	notAllowedError := errors.New("not_allowed")
+
+	// Setup the log in windows events
+	err := el.InstallAsEventCreate(ServiceName, el.Error|el.Info|el.Warning)
+
+	// We expect an error of 'already exists' for subsequent runs,
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return nil, err
+	}
+	il, err := el.Open(ServiceName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Cleanup the handle when exits scope, this could be handled via explicit close but would need to change
+	// more upstream, and its not a huge issue if it hangs around slightly longer than it should
+	runtime.SetFinalizer(il, func(l *el.Log) {
+		l.Close()
+	})
+
+	//  These are setup to be writers for each Windows log level
+	//  Setup this way so we can utilize all the benefits of logformatter
+	infoWriter := &winLogWriter{writer: func(p []byte) {
+		il.Info(1, string(p))
+	}}
+	infoLogger := log.NewLogfmtLogger(infoWriter)
+	infoLogger = level.NewFilter(infoLogger, cfg.LogLevel.Gokit, level.ErrNotAllowed(notAllowedError))
+
+	warningWriter := &winLogWriter{writer: func(p []byte) {
+		il.Warning(1, string(p))
+	}}
+	warningLogger := log.NewLogfmtLogger(warningWriter)
+	warningLogger = level.NewFilter(warningLogger, cfg.LogLevel.Gokit, level.ErrNotAllowed(notAllowedError))
+
+	errorWriter := &winLogWriter{writer: func(p []byte) {
+		il.Error(1, string(p))
+	}}
+	errorLogger := log.NewLogfmtLogger(errorWriter)
+	errorLogger = level.NewFilter(errorLogger, cfg.LogLevel.Gokit, level.ErrNotAllowed(notAllowedError))
+
+	wl := &winLoggerFmt{
+		internalLog:   il,
+		errorLogger:   errorLogger,
+		infoLogger:    infoLogger,
+		warningLogger: warningLogger,
+	}
+	return wl, nil
+
+}
+
+type winLoggerFmt struct {
+	internalLog   *el.Log
+	errorLogger   log.Logger
+	infoLogger    log.Logger
+	warningLogger log.Logger
+}
+
+func (w *winLoggerFmt) Log(keyvals ...interface{}) error {
+	lvl, err := getLevel(keyvals...)
+	// If we don't know what happened move on
+	if err != nil {
+		w.infoLogger.Log(keyvals...)
+		return err
+	}
+	// If the messages level matches one of these we try to write to the logger
+	// the loggers are configured to reject the message if it isn't allowed.
+	if lvl == level.DebugValue() {
+		return w.infoLogger.Log(keyvals...)
+	} else if lvl == level.InfoValue() {
+		return w.infoLogger.Log(keyvals...)
+	} else if lvl == level.ErrorValue() {
+		return w.errorLogger.Log(keyvals...)
+	} else if lvl == level.WarnValue() {
+		return w.warningLogger.Log(keyvals...)
+	}
+
+	return nil
+}
+
+// Looks through the key value pairs in the log for level and extract the value
+func getLevel(keyvals ...interface{}) (level.Value, error) {
+	if len(keyvals)%2 == 1 {
+		keyvals = append(keyvals, nil)
+	}
+	for i := 0; i < len(keyvals); i += 2 {
+		k, v := keyvals[i], keyvals[i+1]
+		if k == "level" {
```

Comment:
The `github.com/go-kit/log/level` package doesn't seem to care about the key, just the value implementing `level.Value`, we should do the same. (It also means the only error we'd have to return is for "no level found", which should default to info-level, as you have now.)